### DISCUSSION TITLE
Context menu grays out from command bind.

### DISCUSSION THREAD

**Discussion by KillzXGaming** [AvaloniaUI/Avalonia#16514] [Q&A]
### Describe the bug

When a MenuItem Command is binded while in a DataTemplate, it seems to gray out and does not work.
Note that all other values bind correctly from the same model, just not the command.

![image](https://github.com/user-attachments/assets/2d84e57a-e72d-4fc3-ba6f-5ed0ba9dc63a)

### To Reproduce

Use xaml

```xml
<Border Background="Gray">
	<Border.ContextMenu>
		<ContextMenu ItemsSource="{Binding ContextMenus}" >
			<ContextMenu.ItemTemplate>
				<DataTemplate>
					<MenuItem Header="{Binding Name}"
									Command="{Binding DoThing}"
									ItemsSource="{Binding Items}"/>
				</DataTemplate>
			</ContextMenu.ItemTemplate>
		</ContextMenu>
	</Border.ContextMenu>
</Border>
```

cs of view model

```cs
    public partial class MainWindowViewModel : ViewModelBase
    {
        private ObservableCollection<ContextMenuItem> _contextMenus = new ObservableCollection<ContextMenuItem>();
        public ObservableCollection<ContextMenuItem> ContextMenus
        {
            get { return _contextMenus; }
            set { SetProperty(ref _contextMenus, value); }
        }

        public MainWindowViewModel()
        {
            this.ContextMenus.Add(new ContextMenuItem("Test"));
        }
    }

    public class ContextMenuItem : ObservableObject
    {
        private string _name = "MenuItem";

        public string Name
        {
            get { return _name; }
            set { SetProperty(ref _name, value); }
        }


        private ObservableCollection<ContextMenuItem> _items = new ObservableCollection<ContextMenuItem>();

        public ObservableCollection<ContextMenuItem> Items
        {
            get { return _items; }
            set { SetProperty(ref _items, value); }
        }

        public ContextMenuItem(string header)
        {
            Name = header;
        }

        public void DoThing()
        {
            Console.WriteLine(this.Name);
        }
    }
```

The same issue occurs if I use a relay command instead.

![image](https://github.com/user-attachments/assets/896522ab-7530-4326-9842-3b3d06aeee84)

### Expected behavior

The menu should display normally and register a proper hit.

### Avalonia version

11.1.1

### OS

Windows

Full app to replicate
[App.zip](https://github.com/user-attachments/files/16404737/App.zip)

**Comment by maxkatz6** [comment]
`DataTemplate` defines content of the MenuItem.
As a result, you are putting MenuItem inside of MenuItem.
I don't know if it's the reason of "graying out", but that's wrong either way. This doc had some example on binding menu items (scroll down) https://docs.avaloniaui.net/docs/0.10.x/controls/menu

**Reply by maxkatz6** [reply]
Also, probably a more important, your command needs to have an `object` parameter. https://docs.avaloniaui.net/docs/guides/data-binding/how-to-bind-to-a-command-without-reactiveui

**Comment by LazaroOnline** [comment]
I have the same issue as @KillzXGaming, he has a point and it does look like a bug, not a missing config on the user end.
Using compiled bindings, it does check that the command is binded to a function in the correct class, so the binding is correct.
However, during runtime, the MenuItem appears greyed out, I don't see any reason for it, and it is not greyed-out if no command parameter is provided.
Here is an example (in a branch):
https://github.com/LazaroOnline/AvaloniaDataGridRowFlyoutRefresh-BugSample/tree/issue-example-contextmenu-greyedout

I created an issue for this with the info here:
https://github.com/AvaloniaUI/Avalonia/issues/18032
